### ISSUE TITLE
Adjusting a views .navigate function

### ISSUE THREAD

**Issue by jgknott** [intljusticemission/react-big-calendar#209]
I've extended the Week view to be a three day view.  Everything works fine, except for the next/previous navigations in the toolbar.  They still adjust by weekly increments even after adjusting the .navigate function to increment by 1 day.

Code below:
```
class ThreeDay extends Week {
  render() {
    let { date } = this.props
    let { start, end } = ThreeDay.range(date, this.props)
    return (
      <TimeGrid {...this.props} start={start} end={end} eventOffset={15} />
    );
  }
}

ThreeDay.navigate = (date, action) => {
  switch (action){
    case navigate.PREVIOUS:
      return dates.add(date, -1, 'day');

    case navigate.NEXT:
      return dates.add(date, 1, 'day')

    default:
      return date;
  }
}

ThreeDay.range = (date, { culture }) => {
  let firstOfWeek = localizer.startOfWeek(culture)
  let start = date
  let end = dates.endOf(date, 'week', firstOfWeek)
  end = dates.add(start, 2, 'day')
  return { start, end };
}
```

What am I missing?

**Comment by jquense** [comment]
try not extending Week in the component?

**Comment by jgknott** [comment]
So write a completely new view? I basically wanted everything the week view has but to only show 3 days and navigate a day at time.  

Can you confirm calling .navigate works as per your example if you change `week` to `day`? https://github.com/intljusticemission/react-big-calendar/blob/master/examples/demos/customView.js

Do I need to call something in the render() function like is called for .range() ?

**Comment by nsearle** [comment]
@jquense - After digging through this for a bit, it seems that the .navigate is not extendable. 

Looking at the `handleNavigate` function in the Calendar component, it is just bypassing the custom components passed in for views. This doesn't allow for any custom `.navigation` to be declared on the originating view component because the nested `moveDate` function is only reading from the default Views and not taking in to account the custom view components. 

I've temporarily fixed this by retrieving the view component from `getView` and passing that component down to the `moveDate` instead of the view string.

**Comment by jquense** [comment]
> So write a completely new view?

yes but you have already mostly done that, there is no other logic in those views except the render and the static methods. Week is just a time grid which a 7 day range

That said u think @nsearle located the bug. care to submit a PR?

**Comment by nsearle** [comment]
@jquense - https://github.com/intljusticemission/react-big-calendar/pull/214

There is a stab at a potential `.navigate` override fix.

**Comment by nsearle** [comment]
@jquense any plan on a release date?
